crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1039,6 +1039,126 @@ impl FromReflect for Cow<'static, str> {
     }
 }
 
+impl<T: FromReflect + Clone + Send + Sync> Array for Cow<'static, [T]> {
+    fn get(&self, index: usize) -> Option<&dyn Reflect> {
+        self.as_ref().get(index).map(|x| x as &dyn Reflect)
+    }
+
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect> {
+        self.to_mut().get_mut(index).map(|x| x as &mut dyn Reflect)
+    }
+
+    fn len(&self) -> usize {
+        self.as_ref().len()
+    }
+
+    fn iter(&self) -> ArrayIter {
+        ArrayIter::new(self)
+    }
+
+    fn drain(self: Box<Self>) -> Vec<Box<dyn Reflect>> {
+        self.iter().map(Reflect::clone_value).collect()
+    }
+}
+
+impl<T: FromReflect + Clone + Send + Sync> Reflect for Cow<'static, [T]> {
+    fn type_name(&self) -> &str {
+        std::any::type_name::<Self>()
+    }
+
+    fn get_type_info(&self) -> &'static TypeInfo {
+        <Self as Typed>::type_info()
+    }
+
+    fn into_any(self: Box<Self>) -> Box<dyn Any> {
+        self
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn into_reflect(self: Box<Self>) -> Box<dyn Reflect> {
+        self
+    }
+
+    fn as_reflect(&self) -> &dyn Reflect {
+        self
+    }
+
+    fn as_reflect_mut(&mut self) -> &mut dyn Reflect {
+        self
+    }
+
+    fn apply(&mut self, value: &dyn Reflect) {
+        crate::array_apply(self, value);
+    }
+
+    fn set(&mut self, value: Box<dyn Reflect>) -> Result<(), Box<dyn Reflect>> {
+        *self = value.take()?;
+        Ok(())
+    }
+
+    fn reflect_ref(&self) -> ReflectRef {
+        ReflectRef::Array(self)
+    }
+
+    fn reflect_mut(&mut self) -> ReflectMut {
+        ReflectMut::Array(self.to_mut())
+    }
+
+    fn reflect_owned(self: Box<Self>) -> ReflectOwned {
+        ReflectOwned::Array(Box::new(self.to_vec()))
+    }
+
+    fn clone_value(&self) -> Box<dyn Reflect> {
+        Box::new(self.clone_dynamic())
+    }
+
+    fn reflect_hash(&self) -> Option<u64> {
+        crate::array_hash(self)
+    }
+
+    fn reflect_partial_eq(&self, value: &dyn Reflect) -> Option<bool> {
+        crate::array_partial_eq(self, value)
+    }
+}
+
+impl<T: FromReflect + Clone + Send + Sync> Typed for Cow<'static, [T]> {
+    fn type_info() -> &'static TypeInfo {
+        static CELL: NonGenericTypeInfoCell = NonGenericTypeInfoCell::new();
+        CELL.get_or_set(|| TypeInfo::Array(ArrayInfo::new::<Self, T>(0)))
+    }
+}
+
+impl<T: FromReflect + Clone + Send + Sync> GetTypeRegistration for Cow<'static, [T]> {
```

Comment:
`ReflectDeserialize` and `ReflectSerialize` require that the type implements `Deserialize` and `Serialize`, respectively. 

```suggestion
impl<'de, T: FromReflect + Clone + Serialize + Deserialize<'de>> GetTypeRegistration for Cow<'static, [T]> {
```

Reply:
However... what if we want to be able to reflect one that isn't serializable? Now that's not possible.

Reply:
I added the ```Deserialize``` and ```Serialize``` trait bounds. Thankfully, I had just read this [post](https://fasterthanli.me/articles/the-curse-of-strong-typing) that talks about higher-ranked trait bounds, because that came in very handy here.

Reply:
I disagree that we should add `Serialize` and `Deserialize` bounds. Again, look at the implementation for `[T; N]`. The reason we don't have those trait bounds there is because it really limits what `T` can be.

`ReflectSerialize` and `ReflectDeserialize` can always be registered manually (like we do with many other types).

Reply:
That makes sense. I changed the ```GetTypeRegistration``` impl to match the other generic types in std.rs.